c10/core/impl/COW.cpp

```diff
@@ -48,7 +51,24 @@ bool is_cow_data_ptr(const c10::DataPtr& data_ptr) {
   return (void*)data_ptr.get_deleter() == (void*)&cow::cow_deleter;
 }
 
-c10::intrusive_ptr<StorageImpl> lazy_clone_storage(StorageImpl& storage) {
+bool is_cowsim_data_ptr(const c10::DataPtr& data_ptr) {
+  return (void*)data_ptr.get_deleter() == (void*)&cow::cowsim_deleter;
+}
+
+c10::intrusive_ptr<StorageImpl> lazy_clone_storage_impl(
+    StorageImpl& storage,
+    bool simulate) {
+  if (get_future_copy_instead_of_conditional_view()) {
```

Comment:
Good point, we can remove the `simulate` arg from here. I'll remove the `simulate_lazy_clone_storage` function as well, and just have `lazy_clone_storage` switch on the global setting.

I feel like we should still have `torch._simulate_lazy_clone` and `torch._lazy_clone` as separate functions, rather than making `torch._lazy_clone` switch on the global setting. Since those functions are only used for testing, I think it's nice to have the tests explicitly call the sim or no-sim version--but they would still need to error if the global setting is not set appropriately